### DISCUSSION TITLE
Apostas para 2018.

### DISCUSSION THREAD

**Discussion by synga** [frontendbr/forum#900] [Discussão]
Fala ai meus queridos Devs desse fórum. Sei que já estamos perto do final desse primeiro mês, mas a enxurrada de posts sobre "2018 trends" que tenho visto é enorme e me motivou a trazer essa discussão pra cá e ver o que cada um espera de 2018 pro front ou pra qualquer área de desenvolvimento ai. É algo legal para saber qual rumo a galera acha que estamos seguindo.

Não poderia trazer isso pra cá sem deixar meu achismo e apostas também, então vamos lá:

**JavaScript** - Apesar do crescimento da popularidade de React, Angular, Vue, Ember e outros creio que a galera vai se ater mais ao básico. A maioria está caindo na real que não adianta SÓ focar nesses frameworks e deixar de melhorar no JS.

**PWA** - Ainda ouço falar pouco por aqui, uma pessoa ou outra que sigo no Twitter solta algo (mas também ainda sigo poucos Devs lá), mas na gringa esse já é um assunto bem falado a um tempo e muita gente lá esta na esperança de que isso tome forças. Ainda tenho uma mentalidade de pensar "Ah mas aqui no Brasil..." e estou tentando mudar isso, o receio é um usuário pensar "Por que vou abrir um site para acessar se posso ter um aplicativo (instalável via play store)?", mas tudo é adaptação e se a tecnologia crescer e os Devs começarem a produzir a galera vai aderir sozinha.

**VueJS** - Em questão de framework creio que o VueJS vai ter um crescimento legal, varias pesquisas apontam o quanto a galera está [querendo aprender](https://stateofjs.com/2017/front-end/results), mas muitos ainda estão receosos devido ao mercado sendo dominado por React e Angular (mesmo que o Angular esteja caindo no conceito dos Devs).

**Web Components** - Componentes HTML reusáveis, acho que não preciso falar mais nada. Da uma olhada em [StencilJS](https://stenciljs.com/).

E vocês? O que esperam pra esse ano?

**Comment by kazzkiq** [comment]
Acredito que uma onda que pode vir forte são os ***compilers-as-frameworks***.

Ao invés de ter uma lib/framework fazendo parse dos seus componentes no browser (como React e Angular fazem, por exemplo), a ideia é compilar todo seu código para JavaScript puro em ambiente de desenvolvimento, reduzindo o número de dependências no browser, aumentando a performance e encurtando o tempo de execução do código.

[Svelte](https://svelte.technology/) e [Stencil](https://stenciljs.com/) (como vc citou) são dois exemplos dessa nova metodologia, e mais compiladores devem surgir em breve.

Pra quem se interessar sobre o assunto, seguem dois links pra dar uma introdução:
https://svelte.technology/blog/frameworks-without-the-framework
https://svelte.technology/blog/sapper-towards-the-ideal-web-app-framework

**Comment by lucas-oliveira-sympla** [comment]
Eu acredito que esse ano seja o inicio dos componentes web( eu utilizo
Polymer para fazer isso!) e PWA!

**Comment by matheusml** [comment]
Oi @synga,

Criei um _post_ recentemente justamente sobre isso, [As 4 tendências pra 2018 que você precisa saber. Ou não.](https://medium.com/tableless/as-4-tend%C3%AAncias-pra-2018-que-voc%C3%AA-precisa-saber-ou-n%C3%A3o-872b1e778580)

A versão resumida do _post_ é:
- PWA
- Criptomoedas
- React
- Privacidade

Além dessas citadas, acho que é bom dar uma olhada em [ReasonML](https://reasonml.github.io/)

Abraços!

**Comment by isaacguerreir** [comment]
Além do que já foi dito acredito que outras tendências importantes:

- WebAssembly
- Outras linguaguens compiladas para WebAssembly (C, Rust, Python)
- Arquitetura de código no client-side, retirando códigos não-UI da Main Thread

**Comment by fernahh** [comment]
Como frameworks/libs, acredito que esse ano as demandas do mercado vão ser React, VueJS e Angular 2+.

Não gosto muito desse termo PWA, mas acredito que cada vez mais será necessário desenvolvedores com grande especialidade em perfomance. Se formos pensar no Brasil, as aplicações web baseadas em JavaScript tem na média de 3, ~4 anos de vida, e agora que elas vão começar a ter problema de escalabilidade. Por esse motivo, acho vai existir bastante demanda por profissionais com essas skills, assim como no back-end existe uma grande demanda por devs especialistas em SRE.

**Comment by yanmagale** [comment]
Grande pergunta @synga  ... 

Com relação a tendências tecnológicas que devem crescer ainda mais:
* VR
* Machine Learning

Sobre os pontos que você levantou:
* PWA: Diria que já é uma realidade e que devemos ver este ano são mais e mais cases. O ano mal começou e tivemos um grande passo na adoção, que em uma das próximas versões do Safari já teremos `Service Worker`

* Web Componentes: Também já é uma realidade. Todos os grandes frameworks/libs do mercado já te oferecem este uso, porém não da forma nativa que o seu browser já te oferece (você pode checar o suporte [aqui](https://www.webcomponents.org/). A APIs melhoraram bastante e tenho certeza que é questão de tempo até que os frameworks/libs do mercado passem a utilizar as features nativas do browser, utilizando o poder que a plataforma já possui (que nem o @lcassiano1612 levantou, também estou fazendo muitas coisas com o Polymer, que já faz isso hoje :smile: )

* Javascript: Sem dúvida alguma. Frameworks/libs vem e vão, mas se você não souber a base, não irá conseguir se adaptar aos novos frameworks que virão.

**Comment by varlen** [comment]
Acrescentando a discussão, as empresas estão começando a notar o valor do **VueJS** devido a entre outros fatores, sua curva de aprendizado suave. Acho que veremos (😆) o Vue aumentar sua presença no mundo mobile através do Weex e do Quasar, quem sabe até mesmo superar o Angular.

**Comment by expalmer** [comment]
E essa nova Lib JS do criador do Ruby on Rails ?

**Comment by milenevlacerda** [comment]
Oii @synga

Acredito que o ano de "bum" de Vue.js e PWA foi o ano passado no sentido de conhecimento e popularização... vi muitas palestras sobre (inclusive dei algumas delas) e acredito que esse ano vai ser de bastante iniciação de implementações com essas tecnologias.

Li [esse](https://medium.com/javascript-scene/top-javascript-libraries-tech-to-learn-in-2018-c38028e028e6) artigo que fala muito sobre tendencias para 2018 e achei bem interessante, indico :) 

Outras tenologias que vejo o pessoal falando bastante é machine learning e blockchain, sei que tem muito hype, mas acredito serem coisas que a galera vai focar em aprender ou consolidar o que sabe sobre.

**Comment by milenevlacerda** [comment]
@yanmagale acredito que PWA ainda não é uma realidade de prática, e sim de conhecimento. Uma galera conhece, mas poucos implementam. Acredito que ano passado uma boa galera começou a olhar pra isso, mas acho que esse ano ela se torna mais popular em implementações... o que tu acha?

**Comment by yunusbra1** [comment]
E quanto ao C++?
